src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,25 +1,39 @@
 package core.mate.academy.service;
 
 import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
 import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
 import core.mate.academy.strategy.BulldozerProducer;
-
+import core.mate.academy.strategy.ExcavatorProducer;
+import core.mate.academy.strategy.TruckProducer;
 import java.util.ArrayList;
-import java.util.Collections;
 import java.util.List;
 
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl implements MachineService<Machine>{
+public class MachineServiceImpl implements MachineService<Machine> {
     @Override
     public List<Machine> getAll(Class type) {
-        List<Machine> machines = new ArrayList<>();
-        return machines;
+        if (type == Bulldozer.class) {
+            BulldozerProducer bulldozerProducer = new BulldozerProducer();
+            List<? extends Machine> machines = bulldozerProducer.get();
+            return new ArrayList<>(machines);
+        } else if (type == Truck.class) {
+            TruckProducer truckProducer = new TruckProducer();
+            List<? extends Machine> machines = truckProducer.get();
+            return new ArrayList<>(machines);
+        } else if (type == Excavator.class) {
+            ExcavatorProducer excavatorProducer = new ExcavatorProducer();
+            List<? extends Machine> machines = excavatorProducer.get();
+            return new ArrayList<>(machines);
+        }
+        return new ArrayList<>();
```

Comment:
По моему тут дублируется код, где ты создаёшь LIst используя в разных наследников интерфейса MachineProducer.
По моему лучше сделать таким образом:
```suggestion
        MachineProducer machineProducer = null;
        if (type == Bulldozer.class) {
            machineProducer = new BulldozerProducer();
        } else if (type == Truck.class) {
            machineProducer = new TruckProducer();
        } else if (type == Excavator.class) {
            machineProducer = new ExcavatorProducer();
        }
        if (machineProducer != null) {
            List<? extends Machine> machines = machineProducer.get();
            return new ArrayList<>(machines);
        }
        return new ArrayList<>();
```

Reply:
Да, по длине почти то же самое, но как по мне, мой вариант смотрится читаемей